### PR TITLE
Move RFCs to the /docs directory

### PR DESCRIPTION
**Pull request by jjulian**
Fulfills the github portions of [the RFC for changing RFCs](https://vfs.atlassian.net/wiki/spaces/~237392673/pages/2120712200/RFC+2022-03-14+Engineering+Decisions+will+be+simplified+to+only+include+RFCs+and+ADRs).

Changes in this PR:
* move existing RFCs to the /docs dir
* improve RFC template
* add a README to help RFC authors understand the process
* add a PR template for RFCs (optional use)
* set CODEOWNERS for the /docs dir to the Platform Engineering Council members

Next steps:
* [ ] update [confluence page](https://vfs.atlassian.net/wiki/spaces/ECP/pages/660373515/When+to+write+ADR+Idea+RFC+or+Design+Doc) to point here
* [ ] migrate in-flight RFC PRs to this dir
* [ ] migrate existing RFCs in Confluence to this repo

All comments welcome. 
- specific question: who should be tagged in CODEOWNERS as reviewers for every RFC? Should we create a github team for that?

### CODE REVIEW

**Comment by CoryTrimmUSDS** [comment]
Forgive my lack of knowledge here - But, by moving the RFCs to the `docs/` top-level folder, are we anticipating that more than just platform engineering will utilize this process?

File: docs/request-for-comment/zz-RFC-template.md [hunk 1]
```diff
@@ -0,0 +1,26 @@
+# RFC: _title_
+_The title is what you want comments on. Use the active voice, and future tense. Example: The website will be built using the React framework_
+* Date: _YYYY-MM-DD_
+* Authors: _tag yourself and other authors here_
```
Comment (mchelen-gov) [inline docs/request-for-comment/zz-RFC-template.md:4]: let's add fields for author's crew and team

File: docs/request-for-comment/zz-RFC-template.md [hunk 2]
```diff
@@ -0,0 +1,26 @@
+# RFC: _title_
+_The title is what you want comments on. Use the active voice, and future tense. Example: The website will be built using the React framework_
+* Date: _YYYY-MM-DD_
```
Comment (mchelen-gov) [inline docs/request-for-comment/zz-RFC-template.md:3]: add field for when comments will be closing

File: docs/request-for-comment/zz-RFC-template.md [hunk 3]
```diff
@@ -0,0 +1,26 @@
...
+* Authors: _tag yourself and other authors here_
+* Status: **PROPOSED**
+* Related Issues: _add links to github/zenhub issues_
+* Related Discussions: _add links to other documents, slack discussions, etc_
+
```
Comment (mchelen-gov) [inline docs/request-for-comment/zz-RFC-template.md:8]: add link to the RFC PR

File: docs/request-for-comment/README.md [hunk 4]
```diff
@@ -0,0 +1,30 @@
...
+# Why do we use RFCs?
+We write an RFC to *communicate intent and get feedback* before we start building something. [RFC driven development](https://engineering-management.space/post/rfc-driven-development/) is a good concept to consider as we build for the Platform.
+
+# Who writes an RFC?
+You! Anyone on a team, including product managers, UX, research, as well as engineers.
+
+# When do I write an RFC?
+Anytime! Whenever your team is making a decision that could affect the platform, write the details and get feedback from others on the Platform.
+
+# How do I create an RFC and manage its lifecycle?
```
Comment (little-oddball) [inline docs/request-for-comment/README.md:15]: In the template, there is a status that is set to PROPOSED, what are the other valid status choices and how does one determine setting from one to the other?

File: docs/request-for-comment/README.md [hunk 5]
```diff
@@ -0,0 +1,30 @@
...
+* create your RFC from this [template](zz-RFC-template.md)
+* name your rfcs file using this format: `YYYY-MM-DD-<definitive statement of your decision>`
+* write!
+* push your branch
+* open a PR
+  - append `template=issue_template.md` to the "compare" url to add the description template (optional - description should be minimal or even blank, all the content should be in the RFC md file itself)
+  - name the PR similar to the title of your RFC: `RFC: YYYY-MM-DD <definitive statement of your decision>`
+* share widely. solicit comments directly from folks you think should be involved
+* resolve comments by updating your RFC file. treat the process like a code PR
+* after all comments are resolved, and all discussion has ended, merge your PR
```
Comment (little-oddball) [inline docs/request-for-comment/README.md:29]: Assumption here is a resolved comment may or may not require an update to the RFC but should be noted it was addressed and what that handling was (discussion w/ no action or what have you).

File: docs/request-for-comment/zz-RFC-template.md (hunk 3 above)
Comment (little-oddball) [reply docs/request-for-comment/zz-RFC-template.md:8]: We want to add a link to itself or am I missing something here?

**Comment by jjulian** [comment]
@td-usds are these comments resolved in the new repo, or do we need to bring them over?

**Comment by andreahewitt-odd** [comment]
@td-usds Could you close this with an explanation and link it up?

**Pull request closed without merging**